### ISSUE TITLE
`view`/`reshape` on GPU drops `requires_grad`

### ISSUE THREAD

**Issue by ngutten** [dollspace-gay/ferrotorch#9]
Calling `.view()` on a GPU tensor with `requires_grad=true` returns a tensor with `requires_grad=false` and no `grad_fn`. This silently breaks the computation graph as there will just be missing gradients.

```rust
let v = Tensor::<f32>::from_storage(
    TensorStorage::cpu(vec![1.0, 2.0, 3.0, 4.0, 5.0, 6.0]),
    vec![2, 3], true,
)?.to(Device::Cuda(0))?;

let v2 = v.view(&[3i64, 2])?;
// v2.requires_grad() == false  ← BUG
// v2 has no grad_fn

let loss = v2.mul_t(&v2)?.sum_all()?;
loss.backward()?;
v.grad() // → None
```

Works correctly on CPU. The issue is in `tensor.rs:view_reshape` which creates a zero-copy view but discards the autograd metadata.
